src/StatusQ/Controls/StatusIdenticonRing.qml

```diff
@@ -0,0 +1,60 @@
+import QtQuick 2.0
+import QtQuick.Layouts 1.0
+
+import StatusQ.Core 0.1
+
+Item {
+    id: identiconRing
+
+    property StatusIdenticonRingSettings settings: StatusIdenticonRingSettings {
+        ringSpec: undefined
+        totalRingUnits: 1
+        initalAngleRad: 0
+        ringPxSize: 1.5
+    }
+
+    readonly property real segmentRadLen: 2 * Math.PI / settings.totalRingUnits
+    // 32 distinctive segment: They are expected to be used using the array index from 0 to 31 and with the order defined in figma (it should be defined as well in specs.)
+    readonly property var distinctiveColors: ["#000000", "#726F6F", "#C4C4C4", "#E7E7E7", "#FFFFFF", "#00FF00",
+                                              "#009800", "#B8FFBB", "#FFC413", "#9F5947", "#FFFF00", "#A8AC00",
+                                              "#FFFFB0", "#FF5733", "#FF0000", "#9A0000", "#FF9D9D", "#FF0099",
+                                              "#C80078", "#FF00FF", "#900090", "#FFB0FF", "#9E00FF", "#0000FF",
+                                              "#000086", "#9B81FF", "#3FAEF9", "#9A6600", "#00FFFF", "#008694",
+                                              "#C2FFFF", "#00F0B6"]
+
+    function printArcSegment(context, xPos, yPos, radius, color, startAngle, arcAngleLen, lineWidth) {
```

Comment:
It does not make much sense to expose it as a public API. Please consider putting this private function into:
```
QtObject {
    id: d
    function printArcSegment(...) {...}
}
```
id: d - detail

or defining it inside `Canvas` component.